src/lib/pages/validator-details/components/bonded-token-changes/BondedTokenChangeMobileCard.tsx

```diff
@@ -1,40 +1,50 @@
-import { Flex, Heading, Text } from "@chakra-ui/react";
+import { Flex } from "@chakra-ui/react";
 
 import { CustomIcon } from "lib/components/icon";
+import { Loading } from "lib/components/Loading";
+import { ErrorFetching } from "lib/components/state";
+import { useValidatorHistoricalPowers } from "lib/services/validatorService";
+import type { AssetInfos, Option, ValidatorAddr } from "lib/types";
+
+import { BondedTokenChangeDetails } from "./BondedTokenChangeDetails";
 
 export const BondedTokenChangeMobileCard = ({
-  denom,
+  validatorAddress,
+  singleStakingDenom,
+  assetInfos,
   onViewMore,
 }: {
-  denom: string;
+  validatorAddress: ValidatorAddr;
+  singleStakingDenom: Option<string>;
+  assetInfos: Option<AssetInfos>;
   onViewMore: () => void;
-}) => (
-  <Flex
-    backgroundColor="gray.900"
-    p={4}
-    rounded={8}
-    w="100%"
-    justifyContent="space-between"
-    alignItems="center"
-    onClick={onViewMore}
-  >
-    <Flex direction="column" gap={2}>
-      <Heading as="h6" variant="h6">
-        Bonded Token Changes
-      </Heading>
-      <Flex alignItems="center" gap={2}>
-        <CustomIcon name="chevron-up" boxSize={6} color="success.main" />
-        <Heading as="h5" variant="h5">
-          7.21%
-        </Heading>
-      </Flex>
-      <Text variant="body1">
-        <Text as="span" fontWeight={700}>
-          +24.02
-        </Text>{" "}
-        {denom} in last 24 hr
-      </Text>
+}) => {
+  const { data: historicalPowers, isLoading } =
+    useValidatorHistoricalPowers(validatorAddress);
+
+  if (isLoading) return <Loading />;
+  if (!historicalPowers) return <ErrorFetching dataName="historical powers" />;
+
+  const assetInfo = singleStakingDenom
+    ? assetInfos?.[singleStakingDenom]
+    : undefined;
+
+  return (
+    <Flex
+      backgroundColor="gray.900"
+      p={4}
+      rounded={8}
+      w="100%"
+      justifyContent="space-between"
+      alignItems="center"
+      onClick={onViewMore}
+    >
+      <BondedTokenChangeDetails
+        historicalPowers={historicalPowers}
+        singleStakingDenom={singleStakingDenom}
+        assetInfo={assetInfo}
+      />
+      <CustomIcon boxSize={6} m={0} name="chevron-right" color="gray.600" />
     </Flex>
-    <CustomIcon name="chevron-right" color="gray.600" />
-  </Flex>
-);
+  );
```

Comment:
what do you think about putting this together with `VotingPowerChart.tsx`?

In `VotingPowerChart.tsx`
```
const isMobile = useMobile();
const isMobileOverview = isMobile && !!onViewMore;
...
if (!historicalPowers) return <ErrorFetching ... />;

if (isMobileOverview) return (
  <Flex
      ... this returned component ...
  </Flex>
);
```

Reply:
`isMobileOverview` is not a prop name though. we already have prop `onOverview` and just call `isMobile` from hook

Reply:
Fixed